docs/DeveloperGuide.rst

```diff
@@ -0,0 +1,182 @@
+
+To ``ACTIVATE`` it::
+
+    $ source ~/PyVENV/bin/activate
+
+To ``DEACTIVATE`` it::
+
+    (PyVENV) $ deactivate
+
+To ``DESTROY`` it::
+
+    (PyVENV) $ deactivate
+    $ rm -Rv ~/PyVENV
+
+You may also want to install Virtualenvwrapper to help manage Virtual Environments although it is not required.
+Reference: 'https://vitualenvwrapper.readthedocs.org/ <https://vitualenvwrapper.readthedocs.org/>'
+
+Clone starcellbio_html
+----------------------
+You may also need to install Homebrew and pip as these are required.
+
+.. note:: This example uses assumes that PROJECT_HOME is ~/starcellbio/starcellbio_html.
+You will need to adjust PROJECT_HOME to the location of your starcellbio_html project.
+
+In a terminal window execute the following commands::
+
+    $ source ~/PyVENV/bin/activate
+    (PyVENV) $ git clone git://github.com/starteam/starcellbio_html
+
+
+Setup Development Environment
+-----------------------------
+
+Using PyCharm
+    View->Tool Windows->Database + Data Source -> MySQL
+    Install the database drivers if prompted
+
+
+mysqld command starts database
+
+Reference: `Installing MySQL on Mac OS X <https://rtcamp.com/tutorials/mac/osx-brew-php-mysql-nginx/>`_
+
+If you need to install MySQL on Mac OS X, execute the following command::
+
+    $ brew install mysql --enable-debugging
+
+In a terminal window execute the following commands::
+
+    $ source ~/PyVENV/bin/activate
+    (PyVENV) $ ls /usr/local/Cellar/mysql # shows foldername (a number) to use in the next command
+    (PyVENV) $ sudo cp /usr/local/Cellar/mysql/5.6.22/homebrew.mxcl.mysql.plist ~/Library/LaunchAgents/
+
+To Start mysql::
+
+    (PyVENV) $ sudo launchctl unload -w ~/Library/LaunchAgents/homebrew.mxcl.mysql.plist
+    (PyVENV) $ sudo launchctl load -w ~/Library/LaunchAgents/homebrew.mxcl.mysql.plist
+
+To Stop mysql::
+
+    (PyVENV) $ sudo launchctl unload -w ~/Library/LaunchAgents/homebrew.mxcl.mysql.plist
+
+To pip install
+
+    (PyVENV) $ cd starcellbio_html
+    (PyVENV) $ sudo pip install -r requirements.txt
+
+
+Using watch.py to compile .soy and .gss files into .js and .css files
+---------------------------------------------------------------------
+After any changes to .soy or .gss files you need to compile them into .js and .css files for your changes to work.
+
+.. note:: This repo was built before PyCharm was the default project IDE. It is now possible to create a File Watcher
+in PyCharm that will automatically run the transpilers as .soy and .gss files are changed.  This will require minor
+modifications to watch.py.
+
+In new terminal window execute the following commands::
+
+    $ source ~/PyVENV/bin/activate
+    (PyVENV) $ cd starcellbio_html
+    (PyVENV) $ sudo pip install -r requirements.txt
+
+    (PyVENV) $ export PROJECT_HOME="/Users/starcellbio/starcellbio_html" # watch.py needs repo at PROJECT_HOME
+    (PyVENV) $ cd $PROJECT_HOME/html_app # watch.py needs repo in this location
+    (PyVENV) $ python watch.py
```

Comment:
We should probably port this to [watchdog](https://pypi.python.org/pypi/watchdog)